connectivity/drivers/ble/FEATURE_BLE/COMPONENT_BlueNRG_2/BlueNrg2HCIDriver.cpp

```diff
@@ -60,7 +60,7 @@
 #define RANDOM_STATIC_ADDRESS_OFFSET    0x80
 #define LL_WITHOUT_HOST_OFFSET          0x2C
 
-#define SPI_STACK_SIZE                  1024
+#define SPI_STACK_SIZE                  2048
```

Comment:
Can you make it dependent on the value of `NDEBUG` ?

```suggestion
#ifdef NDEBUG
#define SPI_STACK_SIZE                  1024
#else
#define SPI_STACK_SIZE                  2048
#endif
```

Reply:
+ add reasoning to the commit message (why we are increasing the value)

Reply:
Change applied.
More memory is needed  due to number of function called and allocations needed  inside BLE thread used for Connectedhomeip provisioning. This is needed to avoid crash due to overflow

Reply:
@0xc0170 Even outside of CHIP, the SPI thread crashes if run with an application compiled without optimization.